### ISSUE TITLE
save_file not working correctly with dompdf.php ?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jkoo7yeha/dompdf#172]
```
I am trying to use the 'output_file' and 'save_file' variables as directed on 
http://code.google.com/p/dompdf/wiki/Usage#Invoking_dompdf_via_the_web 

What I'm aiming to do is to save a file as a pdf on my server, rather than 
downloading it. So instead of using the url: 
dompdf.php?base_path=www%2Ftest%2F&input_file=image.html

I am trying to use: 
dompdf.php?base_path=www%2Ftest%2F&input_file=image.html&output_file=image.pdf&s
ave_file=true

However, instead of saving the file, it gives me a download box again as if I 
was going to the original url, and it's not taking into account the 'save_file' 
variable at all.

I'm using the latest dompdf_0-6-0_beta1 release.

Basically, I'm trying to save a pdf with images onto my server. I can save a 
pdf fine by using the code here: 
http://code.google.com/p/dompdf/wiki/Usage#Using_the_dompdf_class_directly

However, the code in the url above blocks out any images (even local ones), 
which going directly to: 
dompdf.php?base_path=www%2Ftest%2F&input_file=image.html does not.
```

Original issue reported on code.google.com by `stu.tipp...@gmail.com` on 10 Aug 2010 at 12:19

**Comment by GoogleCodeExporter** [comment]
```
The output_file and save_file arguments are forced to dompdf_out.pdf and false 
in the 0.6 release, see here : 
http://code.google.com/p/dompdf/source/browse/trunk/dompdf/dompdf.php#276
I don't know why the output_file was forced, but I understand for save_file. 
Allowing to set it for the web would allow to overwrite possibly any file on 
the system, or at least in the web directory. You need to use the class 
directly to do what you want to do, but we need to fix your problem with 
images. What is the code you use to instanciate dompdf and how are written the 
urls of the images in the document?
```

Original comment by `fabien.menager` on 10 Aug 2010 at 9:44
- Changed title: **save_file not working correctly with dompdf.php ?**

**Comment by GoogleCodeExporter** [comment]
```
Hi Fabien,

thanks for your reply, I missed the atuo false value there, thanks!

I've reset them and made sure that only that directory can be overwritten. It's 
on a private box anyway, so should be secure!

As for the images, with the 'save_file' set to default to 'true' now, it saves 
the file every time perfectly with images in it.

The script that wasn't working with images was here: 
http://code.google.com/p/dompdf/wiki/FAQ under the section 'How do I save a PDF 
to disk?', the code:

[CODE]
require_once("dompdf_config.inc.php");
$html =
    '<html><head><style>body {  color: #7d7a7a;  font-family: 'trebuchet ms', verdana, sans-serif;}</style></head><body><div style="text-align: center; margin-top: 40%;"><img style="width: 75%;" src="images/dompdf_simple.png"/><br/><span>The PHP 5 HTML to PDF converter</span></div></body> </html>';

$dompdf = new DOMPDF();
$dompdf->load_html($html);

$dompdf->render();

// The next call will store the entire PDF as a string in $pdf

$pdf = $dompdf->output();

// You can now write $pdf to disk, store it in a database or stream it
// to the client.

file_put_contents("saved_pdf.pdf", $pdf);
[/CODE]

I've changed the $html value above to include the html code from 
www/test/image.html (which normally displays the images fine when converted to 
a PDF), however with the code above no matter what I enter for the URL of the 
image, it won't display it!

i.e. I've tried:
<img src="dompdf_simple.png"/>
<img src="images/dompdf_simple.png"/>
<img src="test/images/dompdf_simple.png"/>
<img src="www/images/dompdf_simple.png"/>
<img src="create/images/dompdf_simple.png"/>
<img src="http://www.mydomain.com/create/images/dompdf_simple.png"/>

And none of those will find the image.

I'm not sure if that's supposed to be the case, but I thought I'd pop it here 
in case someone else comes across the same problem as me.

Resetting the 'save_file' value to true has fixed my problem now though, so I 
don't need to use the code above anymore!

Thanks.
```

Original comment by `stu.tipp...@gmail.com` on 11 Aug 2010 at 7:23

**Comment by GoogleCodeExporter** [comment]
```
Those links should have read:

<img src="dompdf_simple.png"/>
<img src="images/dompdf_simple.png"/>
<img src="test/images/dompdf_simple.png"/>
<img src="www/test/images/dompdf_simple.png"/>
<img src="create/www/test/images/dompdf_simple.png"/>
<img src="http://www.mydomain.com/create/www/test/images/dompdf_simple.png"/>

Typos!

```

Original comment by `stu.tipp...@gmail.com` on 11 Aug 2010 at 7:25

**Comment by GoogleCodeExporter** [comment]
Original comment by `fabien.menager` on 27 Aug 2010 at 6:01
- Changed state: **Done**

**Comment by GoogleCodeExporter** [comment]
Original comment by `eclecticgeek` on 30 May 2013 at 5:15
- Added labels: **Restrict-AddIssueComment-Commit**
